pkg/cmd/dcos.go

```diff
@@ -9,16 +9,20 @@ import (
 
 	"github.com/dcos/dcos-cli/pkg/config"
 	"github.com/spf13/cobra"
+	"io"
 )
 
 // rootCmd represents the base command when called without any subcommands.
 var rootCmd = &cobra.Command{
 	Use: "dcos",
 }
 
+var dcosConfig *Cluster
```

Comment:
we should avoid global variables in general, I know there are already some but this will be refactored. How about calling `attachedCluster()` at the beginning of the list-provider command into a local variable?